libraries/AP_HAL_Linux/AnalogIn_Navio.cpp

```diff
@@ -0,0 +1,146 @@
+#include <AP_HAL/AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+
+#include "AnalogIn_Navio.h"
+
+#include <cstdlib>
+#include <unistd.h>
+#include <cstdio>
+#include <errno.h>
+
+extern const AP_HAL::HAL& hal;
+
+union pwm_params {
+    char channel[sizeof("XXXX")];
```

Comment:
Here I am not sure, because I don't know how it should work,
but are you aware sizeof("XXXX") will return 5?